docs/extension.md

```diff
@@ -208,4 +207,17 @@ Also, you can register go struct & call it from javascript.
 
 We have exampleapp with comments in exampleapp directory.
 You can also, import github.com/cloudwan/server module and
-have your own RunServer method to have whole custom route written in go.
\ No newline at end of file
+have your own RunServer method to have whole custom route written in go.
+
+# Goext extension (plugin)
+
+You can implement Gohan extension by native go, as an external plugin
+in a shared library.
+
+```yaml
+  extensions:
+  - code: exampleapp_callback
+    code_type: goext
+    id: example
+    path: .*
+```
```

Comment:
You need a step-by-step tutorial to write, build, configure and run goext plugin by using examples. You already write example files. Also, you need a tutorial of testing extensions.